### PR TITLE
remarshal_0_17: init at 0.17.1; pkgs.formats.yaml_1_1: rename from pkgs.formats.yaml

### LINKED ISSUE #354336: Yaml timestamp strings are no longer quoted
### Describe the bug
The recent remarshal update #348098 has changed how yaml timestamp strings are generated. Given the following Nix expression:
```nix
{ time = "22:30:00"; }
```
On 18536bf04cd71abd345f9579158841376fdd0c5a it's generated as
```yaml
time: '22:30:00'
```
Whilst on 4aa36568d413aca0ea84a1684d2d46f55dbabad7 it's generated as
```yaml
time: 22:30:00
```

I generate my Home Assistant config using Nix and this breaks the configuration in various places. For example, Adaptive Lighting expects a string of the form HH:MM:SS for `sunset_time` but, because it's unquoted in the generated yaml, Home Assistant interprets it as a timestamp and converts it to seconds since midnight.

I've tried quoting the timestamps in Nix but this just results in the yaml generator double quoting the strings. 

### Steps To Reproduce
```nix
with import <nixpkgs> { };
let
  before =
    (import (fetchTarball {
      url = "https://github.com/NixOS/nixpkgs/archive/18536bf04cd71abd345f9579158841376fdd0c5a.tar.gz";
    }) { }).formats.yaml { };

  after =
    (import (fetchTarball {
      url = "https://github.com/NixOS/nixpkgs/archive/4aa36568d413aca0ea84a1684d2d46f55dbabad7.tar.gz";
    }) { }).formats.yaml { };

  testCase = {
    time = "22:30:00";
  };
in
pkgs.runCommand "yaml-timestamp-test" { } ''
  cat ${before.generate "before.yaml" testCase} >> $out
  cat ${after.generate "after.yaml" testCase} >> $out
''
```

### Expected behavior
That it's possible to generate the following yaml with Nix:
```yaml
time: '22:30:00'
```

### Notify maintainers

@mweinelt 

### Metadata

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by mweinelt**
YAML 1.1 and 1.2 interpret various keywords and number formats differently, which can lead to unexpected issues when the generated YAML does not match the YAML version parsed by the application.

- Restores `remarshal_0_17`, the last release series to emit YAML 1.1
- Renames `pkgs.formats.yaml` to `pkgs.formats.yaml_1_1`
- Create `pkgs.formats.yaml` as a backwards compatible alias for `pkgs.formats.yaml_1_1`
- Updates test to include explicit YAML 1.1 gotchas, like the Norway-Issue

Optional:
- We could create `pkgs.formats.yaml_1_2` right now, but I'm not sure what the best abstraction would be

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by vcunat** [comment]
BTW, it's possible to produce JSON and that's always acceptable also as YAML 1.2.

**Comment by emilazy** [comment]
> BTW, it's possible to produce JSON, as that's always acceptable also as YAML 1.2.

Well, [almost](https://stackoverflow.com/a/77410478)…

File: pkgs/pkgs-lib/formats.nix [hunk 1]
```diff
@@ -76,10 +76,11 @@ rec {
 
   };
 
-  yaml = {}: {
+  yaml = lib.warn "`pkgs.formats.yaml` is deprecated in favor of explicitly versioned YAML formatters instead. Use `pkgs.formats.yaml_1_1` for YAML 1.1." yaml_1_1;
```
Comment (Stunkymonkey) [inline pkgs/pkgs-lib/formats.nix:79]: just an idea (maybe not a clever one):

have a default-parameter `format ? "1.1` in place and reuse the the existing handle? Then we do not need to rename all existing `formats.yaml`

Comment (mweinelt) [reply pkgs/pkgs-lib/formats.nix:79]: If we had the module system, and I could specify an enum type, then maybe. But we don't so I think this makes the UX worse.

Comment (Stunkymonkey) [reply pkgs/pkgs-lib/formats.nix]: sure. I agree

**Comment by mweinelt** [comment]
Plan is to merge this tonight, since this will unbreak https://github.com/NixOS/nixpkgs/issues/354336.

File: pkgs/by-name/re/remarshal_0_17/package.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    pyyaml
+    rich-argparse
+    ruamel-yaml
+    tomlkit
+    u-msgpack-python
+  ];
+
+  nativeCheckInputs = [ pythonPackages.pytestCheckHook ];
+
+  passthru.updateScript = throw "This package is pinned to 0.17.1 for YAML 1.1 support";
```
Comment (zowoq) [inline pkgs/by-name/re/remarshal_0_17/package.nix:53]: Using a throw here breaks the nixpkgs-update updateScript service, if the intent is just to disable the bot for this package could the disable comment be used instead?

`# nixpkgs-update: no auto update`

Comment (mweinelt) [reply pkgs/by-name/re/remarshal_0_17/package.nix:53]: Sure! Can you send a PR to follow up?

**Comment by yelite** [comment]
Thanks for the PR! This fixes my home assistant setup too, which has error message 
> `expected str for dictionary value @ data['conditions'][0]['state']. Got False`. 

`state: 'on'` got unquoted by the recent remarshal change. In yaml 1.1, unquoted `on` is a boolean value `true`. Just leaving a note here so people who face the same problem can find this.

**Pull request merged**